### DISCUSSION TITLE
Defensive environment parsing

### DISCUSSION THREAD

**Discussion by allenap** [zed-industries/zed#13599] [Feature Requests]
[`load_shell_environment`](https://github.com/zed-industries/zed/blob/45d4de75b3ba57976fa551cc93edb015e96dbe73/crates/project/src/project.rs#L11688-L11752) and [`load_login_shell_environment`](https://github.com/zed-industries/zed/blob/45d4de75b3ba57976fa551cc93edb015e96dbe73/crates/zed/src/main.rs#L811-L861) both load environment variables by spawning the user's shell, printing out the environment (with `/usr/bin/env`), and parsing the result. They both assume that the environment is UTF-8, and that environment variables don't have things like embedded newlines.

For example, given `FOO` defined like so (note the quotes):

```bash
FOO="123
BAR=456"
```

Zed actually sees two variables, `FOO` and `BAR`.

It's not a biggie – most of the time this won't matter – but it's an edge case that could haunt someone down the line.

I have [a proposal for how to fix it](https://github.com/zed-industries/zed/compare/main...allenap:zed:safer-environment-capture). I've only applied the fix to `load_login_shell_environment` for now, but the same pattern can be used for `load_shell_environment` too.

My question is: should I finish this fix and put it up for review?

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
